### ISSUE TITLE
TileEntites after chunk command do not get deleted anymore

### ISSUE THREAD

**Issue by Postremus** [SBPrime/AsyncWorldEdit-Premium#109]
Hello,

I got some problems with tileentities here. They don't seem to get deleted properly in areas i executed a chunk command.

How to reproduce:
1) select an region
2) do `/chunk set 1`
3) do `/chunk set 0`
4) Now, place a sign down in this region, and write something on this sign
5) Remove the sign with a left click
6) Place a new sign at the same location of the old sign, with other or none text. The old text will reappear.

You can also see similar behaviour with chests, droppers, dispenser. Haven't tested more blocks.

I run no other plugins on this server then AWE and WE.

```
AsyncWorldEditInjector version 3.4.3
AsyncWorldEdit version 3.4.5
WorldEdit version 6.1.3;7a097ca
```

Server Version (can be found here https://ci.destroystokyo.com/job/PaperSpigot/900/):

```
This server is running Paper version git-Paper-900 (MC: 1.10.2) (Implementing API version 1.10.2-R0.1-SNAPSHOT)
```

Java version:

```
openjdk version "1.8.0_91"
OpenJDK Runtime Environment (build 1.8.0_91-8u91-b14-3ubuntu1~16.04.1-b14)
OpenJDK 64-Bit Server VM (build 25.91-b14, mixed mode)
```

Thank you for your help!
If anything more is needed, please let me know.

Greetings
Postremus

**Comment by Postremus** [comment]
This is what happens when you do this with a dispenser.
![2016-11-05_12 51 27](https://cloud.githubusercontent.com/assets/1223135/20029812/e2f50b30-a356-11e6-86b1-d91552ad26c9.png)

**Comment by Postremus** [comment]
Any news on this?

**Comment by SBPrime** [comment]
Sorry for keeping quiet. I was able to reproduce the error but I'm not sure what is causing the problem. I'm still trying to fix it.

**Comment by Postremus** [comment]
Digged around the paper source code, found the problem.
You need to set net.minecraft.server.Chunk.j to true, after you call your WrappedChunk.unloadEntities() method.
This field is set to true, once the chunk is loaded. it is set to false, when the chunk is unloaded.
This is vanilla behaviour.
When you do an chunk operation, this field is set to false, once the entites get removed (see chunk.removeEntities()).
This field is however also checked, while removing an tileentitie from the chunk. it prevents the tileentitie to be removed.

I modified the server, and was able to verify this behaviour.

**Comment by SBPrime** [comment]
Thanks for the heads up. I'll fix that in the next version.

**Comment by SBPrime** [comment]
Ok the problem should be fixed in 3.4.7.

Once again thanks for the info. If its not a problem please contact me on skype (sbprime) I would like to have a more private chat with you ;)
